unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombiner.cs

```diff
@@ -124,17 +127,14 @@ public static Output CombineSkinnedMeshes(Matrix4x4[] bindPoses, Transform[] bon
 
             finalMesh.bindposes = bindPoses;
 
-            var boneWeights = AvatarMeshCombinerUtils.ComputeBoneWeights( layers );
-            finalMesh.boneWeights = boneWeights;
+            (NativeArray<byte> bonesPerVertex, NativeArray<BoneWeight1> weights) = AvatarMeshCombinerUtils.ComputeBoneWeights(layers);
+            finalMesh.SetBoneWeights(bonesPerVertex, weights);
```

Comment:
Wouldn't be better to pass `finalMesh` to `AvatarMeshCombinerUtils.ComputeBoneWeights` instead of returning the couple?

Reply:
It's intended for testing that this method doesn't have a side effect (assigning the bone weights to a mesh), but I've separated this method into 2 to avoid the usage of a tuple.